File: 躲藏.cpp
#include <bits/stdc++.h>
using namespace std;

#define endl '\n'
#define int long long
const int mod = 2000120420010122;

string s;

signed main() {
    cin.tie(nullptr)->sync_with_stdio(false);
    cout << fixed << setprecision(15);
    while (cin >> s) {
        int dpc = 0, dpcw = 0, dpcwb = 0, ans = 0;
        for (char &c : s) {
            c = tolower(c);
            if (c == 'c') {
                dpc += 1;
                ans += dpcwb;
            } else if (c == 'w') {
                dpcw += dpc;
            } else if (c == 'b') {
                dpcwb += dpcw;
            }
        }
        cout << ans % mod << endl;
    }
    return 0;
}

File: 躲藏.py
import sys
input = lambda: sys.stdin.readline().strip()
mod = 2000120420010122

while True:
    try:
        s = list(input())
        if s == []:
            break
    except EOFError:
        break
    dpc, dpcw, dpcwb, ans = 0, 0, 0, 0
    for c in s:
        if c in ['c', 'C']:
            dpc += 1
            ans += dpcwb
        elif c in ['w', 'W']:
            dpcw += dpc
        elif c in ['b', 'B']:
            dpcwb += dpcw
    print(ans % mod)
